### ISSUE TITLE
Adjust text to diagonal and vertical lines

### ISSUE THREAD

**Issue by jcruz-ferreyra** [roboflow/supervision#319]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

I've noticed that the LineZoneAnnotator works wonderfully for horizontal lines. However, when it comes to diagonal or vertical lines, I've found that the "count" information remains aligned vertically with respect to the center of the LineZone. Additionally, if an horizontal line is drawn laterally-inverted (with the start point to the right and end point to the left), the "In" count still appears over the line, even though it should ideally be positioned under it.

For a clearer understanding, I've created an illustration comparing the original behavior to the improved version I've developed:

*Original Behavior:*

![original](https://github.com/roboflow/supervision/assets/105461618/7dcbba4e-ca10-47ad-943f-13d08f4e7f15)

*Improved Behavior:*

![edited](https://github.com/roboflow/supervision/assets/105461618/4e3c1ec4-2050-4364-b9e4-66ee1e350bf4)

In my own project, I needed support for different orientations and directions of LineZones. As a result, I forked the repository and developed a modified version of the LineZoneAnnotator class. In this version, text boxes are right-aligned to the end point of the LineZone, while text orientation is aligned with the direction of it.

I believe these improvements could enhance the usability and flexibility of the LineZoneAnnotator class for users who require support for different line orientations and directions.

Please let me know your thoughts on this proposed change. I would be more than happy to collaborate further on refining and integrating these improvements into the main repository.

### Additional

pd: please ignore changes in the image that are not related to this issue, such as removing the circles at the beginning and end of each line.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by hardikdava** [comment]
@jcruz-ferreyra Thank you for suggesting new features and taking interest in `supervision`. We are already working on improving annotators and going to introduce new annotators as well. We will consider your thoughts on this.

**Comment by SkalskiP** [comment]
Hi @jcruz-ferreyra! I love it! 🔥 At first, I thought the second image was just 'photoshopped'. The fact you made it so configurable blows my mind. We need to have that!

As @hardikdava said, we already started to work on an updated version of the line zone. But I think what you made aligns with our goals. 

We are a little bit overwhelmed by the amount of work right now. Will you give us a few weeks to get back to this conversation? 🙏🏻

**Comment by jcruz-ferreyra** [comment]
Hi @SkalskiP no problem at all! I understand you're busy. Feel free to reach out whenever you're ready to continue. Looking forward to collaborating! 🚀

**Comment by hardikdava** [comment]
@jcruz-ferreyra can you provide us a code snippet? Or feel free to open a PR.

**Comment by jcruz-ferreyra** [comment]
Certainly! I'll create a new fork focusing in the discussed issue. I've personalized a few other things as well, so I'll ensure the PR is clean and specific to the current topic. Expect the PR next week.

**Comment by maddust** [comment]
@jcruz-ferreyra  this is great! I was thinking of implementing something similar, like adding arrows aligned with the line zone counters

**Comment by hardikdava** [comment]
@jcruz-ferreyra please make a PR in branch: [feature/redesigned_annotators](https://github.com/roboflow/supervision/tree/feature/redesigned_annotators). Currently, we are adding/modifying improved annotators related functions there. Thank you for taking time to improve supervision.

**Comment by SkalskiP** [comment]
@maddust I knew you'd love it as well!

**Comment by SkalskiP** [comment]
@jcruz-ferreyra / @hardikdava, any progress I should know about?

**Comment by hardikdava** [comment]
@SkalskiP i haven't start working on it yet.

**Comment by jcruz-ferreyra** [comment]
Done!

**Comment by tonmoy-TS** [comment]
Hey, I was wondering if this feature has been added ?

**Comment by tonmoy-TS** [comment]
@SkalskiP yes, very much. That would be a useful feature to have for any sort of diagonal line. 

I am thinking, another useful feature might be having options to place custom texts (other than the 'in' and 'out' for counting) next to the line zones, and may also have a separate 'id' for each of the line zones (the same goes for polygon zones as well). 
Example use cases: if we have a lot of line zones in the frame and I want to identify them by just looking at the 'custom text' or by the 'id'.  Or, if we want to filter out the detections based on them crossing specific line zones 'id's etc.
 
![custom texts](https://github.com/roboflow/supervision/assets/70789929/8f16f1e4-1c7a-4c8e-85d8-b113cff7bedf)
or
![custom ids](https://github.com/roboflow/supervision/assets/70789929/497724d4-2406-44c9-83b3-9dd9624ece73)

@jcruz-ferreyra , Sorry for using your images without your permission.

**Comment by jcruz-ferreyra** [comment]
Hi @tonmoy-TS, no problem at all. There is already a PR opened with this feature [here](https://github.com/roboflow/supervision/pull/854). You can check the changes in `line_zone.py` file.

Regarding adding an ID to each line, personally, I store them in a dictionary in the implementation using the key as the ID.

Sorry if I'm answering questions you didn't ask; I thought my approach might be useful for you. Hope this helps.
